src/orchard/inspect.clj

```diff
@@ -356,18 +374,69 @@
         (render-ln))
     inspector))
 
+(defn- rendered-element-str
+  "Return the string representation of the rendered `element`."
+  [element]
+  (cond (string? element)
+        element
+        (and (seq? element) (= :value (first element)))
+        (second element)
+        (and (seq? element) (= :newline (first element)))
+        "\n"
+        :else
+        (throw (ex-info "Unsupported element" {:element element}))))
+
+(defn- render-map-key
+  "Render the key of a map."
+  [inspector key]
+  (render-value inspector key))
+
+(defn- long-map-key?
+  "Returns true of `s` is a long string, more than 20 character or
+  containing newlines."
+  [^String s]
+  (or (.contains s "\n") (> (count s) 20)))
+
+(defn- render-map-separator
+  "Render the map separator according to `rendered-key`. If
+  `rendered-key` is long or contains newlines the key and value will
+  be rendered on separate lines."
+  [{:keys [pretty-print] :as inspector} long-key?]
+  (if (and (= "true" pretty-print) long-key?)
+    (-> (render-ln inspector)
+        (render-indent "=")
+        (render-ln))
+    (render inspector " = ")))
+
+(defn- render-map-value
+  "Render a map value. If `mark-values?` is true, attach the keys to the
+  values in the index."
+  [{:keys [pretty-print] :as inspector} key val mark-values? rendered-key long-key?]
+  (if (= "true" pretty-print)
+    (let [indentation (if long-key? 0 (+ 3 (count rendered-key)))]
+      (-> (indent inspector indentation)
+          (render (if (zero? indentation) "  " ""))
+          (render-value val
+                        (when mark-values?
+                          {:value-role :map-value, :value-key key}))
+          (unindent indentation)
+          ((if (long-map-key? rendered-key) render-ln identity))))
+    (render-value inspector val
+                  (when mark-values?
+                    {:value-role :map-value, :value-key key}))))
+
 (defn- render-map-values
   "Render associative key-value pairs. If `mark-values?` is true, attach the keys
   to the values in the index."
   [inspector mappable mark-values?]
   (reduce (fn [ins [key val]]
-            (-> ins
-                (render-indent)
-                (render-value key)
-                (render " = ")
-                (render-value val (when mark-values?
-                                    {:value-role :map-value, :value-key key}))
-                (render-ln)))
+            (let [rendered-key (rendered-element-str (last (:rendered (render-map-key ins key))))
```

Comment:
Still not quite what I meant:) You can print the key with `print-string` first. That gives you a string that you can check if it's long or has newlines and you don't have to dig it out of `:rendered` list and parse it. And once you have a printed string, you can pass it to `render-value` via `:display-value`.